scripts/install

```diff
@@ -78,17 +77,91 @@ ynh_script_progression --message="Storing installation settings..." --weight=1
 
 ynh_app_setting_set --app=$app --key=domain --value=$domain
 ynh_app_setting_set --app=$app --key=path --value=$path_url
-ynh_app_setting_set --app=$app --key=no_sso --value true
 ynh_app_setting_set --app=$app --key=server_name --value=$server_name
 ynh_app_setting_set --app=$app --key=jitsi_server --value=$jitsi_server
 ynh_app_setting_set --app=$app --key=final_path --value=$final_path
 ynh_app_setting_set --app=$app --key=synapse_version --value=$upstream_version
-ynh_app_setting_set --app=$app --key=is_free_registration --value=$is_free_registration
 ynh_app_setting_set --app=$app --key=report_stats --value=$report_stats
-ynh_app_setting_set --app=$app --key=allow_public_rooms --value=$allow_public_rooms
 ynh_app_setting_set --app=$app --key=e2e_enabled_by_default --value=$e2e_enabled_by_default
 ynh_app_setting_set --app=$app --key=synapse_user_app_pwd --value=$synapse_user_app_pwd
 
+if [ $is_free_registration -eq 0 ]
+then
+    enable_registration="false"
+    turn_allow_guests="false"
+    sso_enabled="true"
+    password_enabled="false"
+else
+    enable_registration="true"
+    turn_allow_guests="true"
+    sso_enabled="false"
+    password_enabled="true"
+fi
+
+ynh_app_setting_set --app=$app --key=enable_registration --value=$enable_registration
+ynh_app_setting_set --app=$app --key=turn_allow_guests --value=$turn_allow_guests
+ynh_app_setting_set --app=$app --key=sso_enabled --value=$sso_enabled
+ynh_app_setting_set --app=$app --key=password_enabled --value=$password_enabled
+
+#=================================================
+## SET STANDARD SETTINGS FROM DEFAULT CONFIG
+#=================================================
+
+element_ynh_url="https://matrix.to/"
+# Get app name of first Element Instance (can be changed later in Config Panel)
+element_instance="element"
+if yunohost --output-as plain app list | grep -q "^$element_instance$"; then
+    element_domain=$(ynh_app_setting_get --app $element_instance --key domain)
+    element_path=$(ynh_app_setting_get --app $element_instance --key path)
+    element_ynh_url="https://""$element_domain""$element_path"
+    #if [ -z "$element_domain" ]; then
+    #    element_ynh_url="https://matrix.to/"
+fi
+
+web_client_location=$element_ynh_url
+client_base_url=$element_ynh_url
+invite_client_location=$element_ynh_url
+
+backup_before_upgrade="true"
+server_statistics="false"
+allow_public_rooms_without_auth="false"
+allow_public_rooms_over_federation="false"
+max_upload_size="10M"
+disable_msisdn_registration="true"
+registrations_require_3pid="email"
+# here we need sed magic to transform $server_name
+allowed_local_3pids="'^[^@]+@""matrix""\.org$'"
```

Comment:
here we need to accept I think all domain of the server instead of matrix.org

Reply:
could be a solution yes! But I don't know `sed`.. @thardev ?

Reply:
solved thanks to @rosbeef